src/app/parcels/batch/BatchParcelDataGrid.tsx

```diff
@@ -180,39 +188,212 @@ const styledBatchGridDisplayColumns: GridColDef<BatchGridDisplayRow[][number]>[]
     });
 
 interface BatchParcelDataGridProps {
-    rows: BatchGridDisplayRow[];
+    initialTableState: BatchTableDataState;
 }
 
-export const defaultData: BatchGridDisplayRow[] = Array.from({ length: 2 }, (_, index) => ({
-    id: index,
-    rowName: index,
-    fullName: "",
-    phoneNumber: "",
-    address: "",
-    adults: null,
-    children: null,
-    listType: "",
-    dietaryRequirements: "",
-    feminineProducts: "",
-    babyProducts: "",
-    petFood: "",
-    otherItems: "",
-    deliveryInstructions: "",
-    extraInformation: "",
-    attentionFlag: "",
-    signpostingCall: "",
-    notes: "",
-    voucherNumber: "",
-    packingDate: "",
-    packingSlot: "",
-    shippingMethod: "",
-    collectionInfo: "",
-}));
-
-const BatchParcelDataGrid: React.FC<BatchParcelDataGridProps> = ({ rows }) => {
+const getEmptyRow = (id: number): BatchGridDisplayRow => {
+    return {
+        id: id,
+        rowName: id,
+        fullName: "",
+        phoneNumber: "",
+        address: "",
+        adults: null,
+        children: null,
+        listType: "",
+        dietaryRequirements: "",
+        feminineProducts: "",
+        babyProducts: "",
+        petFood: "",
+        otherItems: "",
+        deliveryInstructions: "",
+        extraInformation: "",
+        attentionFlag: "",
+        signpostingCall: "",
+        notes: "",
+        voucherNumber: "",
+        packingDate: "",
+        packingSlot: "",
+        shippingMethod: "",
+        collectionInfo: "",
+    };
+};
+
+const addressToString = (address: Address | null): string | null => {
+    if (!address) {
+        return null;
+    }
+    const addressArray: string[] = [];
+    for (const key in address) {
+        if (address[key] !== null) {
+            addressArray.push(address[key]);
+        }
+    }
+    return addressArray.join(", ");
+};
+
+const collectionInfoToString = (collectionInfo: CollectionInfo): string => {
+    const { collectionDate, collectionSlot, collectionCentreId } = collectionInfo;
+    return `${collectionDate}, ${collectionSlot}, ${collectionCentreId}`;
+};
+
+const booleanGroupToString = (booleanGroup: BooleanGroup): string => {
+    const trueKeys: string[] = Object.keys(booleanGroup).filter((key) => booleanGroup[key]);
+    return trueKeys.join(", ");
+};
+
+const overrideDataToOverrideDisplayRow = (dataRow: OverrideDataRow): BatchGridDisplayRow => {
```

Comment:
There's a lot of helper functions in this file, can we move them to a `displayHelpers.ts` file (or something like that) to keep it readable?